modulemd/modulemd-module-stream.c

```diff
@@ -968,34 +1015,14 @@ stream_expansion_helper (ModulemdDependencies *deps,
        * and the previous iterations of module:stream combinations currently
        * present in expanded_deps.
        */
-      for (guint j = 0; j < g_strv_length (streams); j++)
+      for (j = 0; j < g_strv_length (streams); j++)
         {
           stream = streams[j];
           g_debug ("Expansion: looking at %s stream dependency %s:%s",
                    which,
                    module,
                    stream);
 
-          /*
-           * If a stream name begins with a '-' sign (which corresponds to
-           * "modulename: [-streamname]" in the spec), the intention is to
-           * exclude this stream from the list of all active streams for this
-           * module. Unfortunately, that is something only the Module Build
-           * Service can do, so we must fail the stream expansion.
-           */
-          if (stream[0] == '-')
-            {
-              g_set_error (error,
-                           MODULEMD_ERROR,
-                           MMD_ERROR_UPGRADE,
-                           "Cannot expand module %s dependency %s using "
-                           "stream exclusion (%s).",
-                           which,
-                           module,
-                           stream);
-              return FALSE;
-            }
```

Comment:
If there was no help from the helper, stream exclusions still need to be caught, reported, and failed. (Although I had overlooked that restriction that if the first stream began with `-` they all do, so this check could be done prior to the loop through the streams.)

Reply:
I'm not sure what's wrong with how it happens now. After the helper does (or doesn't) process it, I check:
```C
      /* Unfortunately, without a set of known_streams, we must fail the stream
       * expansion.
       */
      if (g_strv_length (streams) == 0 || streams[0][0] == '-')
        {
          g_set_error (error,
                       MODULEMD_ERROR,
                       MMD_ERROR_UPGRADE,
                       "Cannot expand module %s dependency %s for all active "
                       "existing streams.",
                       which,
                       module);
          return FALSE;
        }
```

That reads "If the stream list contains no entries or the first character of the first entry is `-`, return an error.

Do you see a case I missed?

Reply:
As I was reviewing the PR, I am positive the github UI was showing me `if (g_strv_length (streams) == 0)` for the conditional on that error message. Looks good.